### PR TITLE
[Network] DNS Fixes

### LINKED ISSUE #1718: [DNS] Record set list doesn't display type name by default
It would be useful if the default table output of the `az network dns record-set list` command displayed the type (e.g. `CNAME`). As it stands now, there isn't a way to easily differentiate between two record sets that share the same name but different type.

Additionally, since the type is a required parameter for the update/delete/show commands for a record set instance, it seemingly represents a key part of the record set's identity, and would useful to be able to easily retrieve without requiring the use of a custom format/query.

### LINKED ISSUE #1719: [DNS] Could record sets be implicitly created when adding new records?
I may be missing something, but it doesn't seem like the `az network dns record <TYPE> add` commands allow implicitly creating a record set if it doesn't already exist. [This behavior](https://docs.microsoft.com/en-us/azure/dns/dns-operations-recordsets-cli#create-a-dns-record) existed in the previous xplat CLI, and would help simplify the experience for common cases where there aren't multiple records in a zone with the same type and name.

Right now, you'd have to run `az network dns record-set create`, and then go create your actual record, which feels a little heavy weight for simple scenarios.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #1718. Fixes #1719. Addresses parts of #1131.

- Automatically creates a record set when using `az record _ add` if the set does not already exist. (#1719)
- Adds the ability to filter `dns record-set list` by `--type`. (#1131)
- **BC**: Removes the `--top` parameter from `dns record-set list`. (#1131)
- Adjusts the table format of `dns record-set list` (#1131) (#1718)

### CODE REVIEW

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -562,3 +560,7 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 
 for item in ['list', 'show']:
     register_cli_argument('network dns record-set {}'.format(item), 'record_type', options_list=('--type', '-t'), **enum_choice_list(RecordType))
+
+for item in ['a', 'aaaa', 'cname', 'mx', 'ns', 'ptr', 'srv', 'txt']:
+    register_cli_argument('network dns record {} add'.format(item), 'record_set_name', options_list=('--record-set-name',), help='The name of the record set relative to the zone. Creates a new record set if one does not exist.')
```
Comment (lostintangent) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: Any chance we could also add a short name alias with this change? Perhaps `-s`?

**Comment by derekbekoe** [comment]
@tjprescott Can you provide examples of the new table output?

**Comment by tjprescott** [comment]
@derekbekoe 
(env) c:\Users\trpresco\Documents\github\azure-cli>az network dns record-set list -g tjp-dns -z myzone.zone -otable
Name    ResourceGroup       Ttl  Type    Metadata
------  ---------------  ------  ------  ----------
@       tjp-dns          172800  NS
@       tjp-dns            3600  SOA
txtset  tjp-dns            3600  TXT

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py [hunk 2]
```diff
@@ -23,6 +23,18 @@ def _strip_null_records(item):
...
+        table_row = OrderedDict()
+        table_row['Name'] = item['name']
+        table_row['ResourceGroup'] = item['resourceGroup']
+        table_row['Ttl'] = item['ttl']
+        table_row['Type'] = item['type'].rsplit('/', 1)[1]
+        table_row['Metadata'] = item['metadata'] or ' '
+        table_output.append(table_row)
+    return table_output
+
 def transform_local_gateway_table_output(result):
```
Comment (troydai) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py]: PEP8: two blank lines

**Comment by tjprescott** [comment]
@derekbekoe: added a change to allow actual display of the metadata (due to the dictionary thing). Here's what the output looks like:
```
Name    ResourceGroup       Ttl  Type    Metadata
------  ---------------  ------  ------  -------------------------------------------------
@       tjp-dns          172800  NS
@       tjp-dns            3600  SOA
txtset  tjp-dns            3600  TXT     foo="doo" noodle="pon" user="Nugget in a biscuit"
```

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py [hunk 3]
```diff
@@ -25,13 +25,18 @@ def _strip_null_records(item):
...
     for item in result:
         table_row = OrderedDict()
         table_row['Name'] = item['name']
         table_row['ResourceGroup'] = item['resourceGroup']
         table_row['Ttl'] = item['ttl']
         table_row['Type'] = item['type'].rsplit('/', 1)[1]
-        table_row['Metadata'] = item['metadata'] or ' '
+        metadata = item['metadata']
+        if metadata:
+            table_row['Metadata'] = ' '.join(['{}="{}"'.format(x, metadata[x]) for x in sorted(metadata.keys())])
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py]: Is `.keys()` redundant in `sorted(...)`?

```
>>> a = {'a': 1, 'b': 2, 'c': 3, 'd': 4}
>>> sorted(a)
['a', 'b', 'c', 'd']
>>> a.keys()
['a', 'c', 'b', 'd']
>>> sorted(a.keys())
['a', 'b', 'c', 'd']
>>> 
```

**Pull request merged**